### PR TITLE
feat: middle SP z selection as result of grid binning

### PR DESCRIPTION
**Pull request by CarloVarni**
This is trying to simplify the middle space point selection process during seeding. As of now we define a z-range validity for the middle space point candidate and when we loop on all the middle candidates we check the value of the z coordinate.

This will do the same, but the selection on the z coordinate would be a consequence of the grid binning.

In other words, this adds two new grid bins in the z dimension (covering the range `2700 < |z| < 3000 mm`). These bins will be skipped during grid iteration as middle candidates.

Other parameters are adapted accordingly in order to return the same seeds as before.

Tagging @noemina @LuisFelipeCoelho @paulgessinger and @andiwand 

Eventually, we may want to do the same for the r-coordinate and completely remove the checks for the middles space point validity

### CODE REVIEW

**Review by andiwand — COMMENTED** [review]
No objections from my side - pinging @LuisFelipeCoelho

File: Examples/Python/python/acts/examples/itk.py [hunk 1]
```diff
@@ -390,32 +393,35 @@ def itkSeedingAlgConfig(
...
-            6,
-            5,
             7,
+            6,
+            8,
         ]  # enable custom z looping when searching for SPs, must contain numbers from 1 to the total number of bin in zBinEdges
         zBinNeighborsTop = [
             [0, 0],
             [-1, 0],
+            [-2, 0],
```
Comment (LuisFelipeCoelho) [inline Examples/Python/python/acts/examples/itk.py:411]: why [-2, 0] and not [-1, 0] ?

File: Examples/Python/python/acts/examples/itk.py [hunk 2]
```diff
@@ -390,32 +393,35 @@ def itkSeedingAlgConfig(
         interactionPointCut = True
         impactMax = 2 * u.mm
         zBinsCustomLooping = [
-            1,
             2,
             3,
             4,
+            5,
+            12,
```
Comment (LuisFelipeCoelho) [inline Examples/Python/python/acts/examples/itk.py:400]: why do you change `zBinsCustomLooping`?

**Comment by LuisFelipeCoelho** [comment]
I only added two questions, but besides that it looks good to me

File: Examples/Python/python/acts/examples/itk.py (hunk 1 above)
Comment (CarloVarni) [reply Examples/Python/python/acts/examples/itk.py:411]: What this PR does is taking the first bin #1 and diverse it in two bins #1 and #2. 

All the other bins then also increase by 1.

Now, in order to keep the same results as before, we need the former bin #2 (which now is #3) to search the top candidate in the two bins on the left. Thus the -2.

The same reasoning applies to the rightmost side of the detector

File: Examples/Python/python/acts/examples/itk.py (hunk 2 above)
Comment (CarloVarni) [reply Examples/Python/python/acts/examples/itk.py:400]: Just increased by 1 all the numbers due to the addition of a bin in the leftmost part of the detector

Comment (LuisFelipeCoelho) [reply Examples/Python/python/acts/examples/itk.py:400]: ah! off course!

File: Examples/Python/python/acts/examples/itk.py (hunk 1 above)
Comment (LuisFelipeCoelho) [reply Examples/Python/python/acts/examples/itk.py:411]: Thanks! I see

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [6c943bf2935a9f8e807b56b6ba421c1689a62301]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7570084)!

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078822)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078820)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078818)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078816)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078815)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078813)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078811)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078809)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078807)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078806)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078804)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078802)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078800)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078798)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078795)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078794)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078791)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078790)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078786)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078782)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078780)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078778)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078777)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078775)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078773)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078771)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078770)||
|:red_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078769)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078767)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078766)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40078765)||

**Pull request merged**